packages/react-graphql/src/hooks/mutation.ts

```diff
@@ -0,0 +1,29 @@
+import {OperationVariables} from 'apollo-client';
+import {MutationOptions} from 'react-apollo';
+import {DocumentNode} from 'graphql-typed';
+import {useMemo} from 'react';
+
+import {MutationHookOptions, MutationHookResult} from './types';
+import {useApolloClient} from './apollo-context';
+
+export default function useMutation<Data = any, Variables = OperationVariables>(
+  mutation: DocumentNode<Data, Variables>,
+  options: MutationHookOptions<Data, Variables> = {},
+): MutationHookResult<Data, Variables> {
+  const {client: overrideClient, ...mutationOptions} = options;
+
+  const client = useApolloClient(overrideClient);
+
+  const runMutation = useMemo(
+    () => (perMutationOptions?: MutationOptions<Data, Variables>) => {
+      return client.mutate({
+        mutation,
+        ...mutationOptions,
+        ...perMutationOptions,
+      });
```

Comment:
Should we do a deep merge here instead of a shallow merge? I think things like `variables` are nested objects, do we think people will pass a subset of options?

Reply:
funny enough, when I was playing with the example, I wanted to...

Reply:
I didn't do a full deep merge but I did carry over what apollo did, which is one level deeper for variables.

https://github.com/apollographql/react-apollo/blob/master/src/Mutation.tsx#L222-L223